src/v/redpanda/application.cc

```diff
@@ -1255,7 +1255,10 @@ void application::start_redpanda(::stop_signal& app_signal) {
     _co_group_manager.invoke_on_all(&kafka::group_manager::start).get();
 
     syschecks::systemd_message("Starting controller").get();
-    controller->start().get0();
+    controller->start(app_signal.abort_source()).get0();
+
+    _deferred.emplace_back([this] { controller->shutdown_input().get(); });
```

Comment:
was reordering of the input shutdown to occur after group migration intentional?

Reply:
Yes, see commit message -- this looked to me like it was an accident in c96745297d where the group migration stuff was added between controller start and registering the controller shutdown hook.

Reply:
oh shoot, my bad.

yes, you are right about https://github.com/redpanda-data/redpanda/blame/dev/src/v/redpanda/application.cc#L1268-L1273 i think that comment can be retained. that commit you referenced seems to have hijacked it.

Reply:
you're right, I've moved the line back to the correct side of the comment